include/envoy/network/address.h

```diff
@@ -142,6 +159,21 @@ class Instance {
    */
   virtual const Ip* ip() const PURE;
 
+  /**
+   * @return the pipe address information IFF type() == Type::Pipe, otherwise nullptr.
+   */
+  virtual const Pipe* pipe() const PURE;
+
+  /**
+   * @return the underlying structure wherein the address is stored
+   */
+  virtual const sockaddr* sockAddr() const PURE;
```

Comment:
I think the interface will be nicer if `sockAddr()` and `sockAddrLen()` are combined into a single function that returns a std::pair.

Reply:
Agreed. Should we do it in this PR or in a different one? The current change was pretty trivial because I just moved the functions from `InstanceBase` to `Instance`. Making the return value a pair would be somewhat more invasive.

Reply:
Either way is fine.

Reply:
+1 either way is fine.

Reply:
Thanks! I'll add it to my todo list.